templates/_helpers.tpl

```
@@ -207,9 +207,9 @@ Params:
   - default - String - Required - Default value if, there is no secret reference under secretRef
 */}}
 {{- define "secrets.key" -}}
-{{- if and .secretRef (not (kindIs "string" .secretRef)) .secretRef.valueFrom.secretKeyRef.key -}}
+{{- if and .secretRef (not (kindIs "string" .secretRef)) -}}
 {{ .secretRef.valueFrom.secretKeyRef.key }}
 {{- else -}}
 {{ .default }}
-{{- end -}}  
+{{- end -}}
```

Add else where you can throw an error that provided value is not correct.